docs/rules/jsx-no-namespace.md

```diff
@@ -0,0 +1,29 @@
+# Enforce that namespaces are not used in JSX (react/jsx-no-namespace)
+
+Enforces the absence of a namespace in JSX components, such as with `svg:circle`, as they are not supported in React.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+<ns:TestComponent />
+```
+
+```jsx
+<Ns:TestComponent />
+```
+
+The following patterns are **not** considered warnings:
+
+```jsx
+<TestComponent />
+```
+
+```jsx
+<testComponent />
+```
+
+## When not to use
+
+If you are not using JSX.
```

Comment:
this seems like it shouldn't be jsx-specific, since it can still apply to `React.createElement` when called with a string literal.

Reply:
I am willing to make the linting rule work with `React.createElement`.
It wouldn't be called `jsx-no-namespace` anymore though. Do you have a suggestion for a name?

Reply:
I think just `no-namespace` would work?